### ISSUE TITLE
TFTP Distinct TFTP Traversal Bug

### ISSUE THREAD

**Issue by unKn0wnUser453** [rapid7/metasploit-framework#13359]
**Module**
use exploit/windows/tftp/distinct_tftp_traversal
set DEPTH 0
set RHOST <IP Address>
set LHOST tun0
run

**Problem**
The payload does not transfer fully and fails.
The depth input is being overwritten.

**Edits Suggested**
/usr/share/metasploit-framework/modules/exploits/windows/tftp/distinct_tftp_traversal.rb
Change Line 74 to: select(nil, nil, nil, 5)
Change Line 89 to: depth = datastore['DEPTH']

**Comment by bcoles** [comment]
The issue you've encountered is probably unrelated to the depth. You're correct that the depth is overwritten, but it is unlikely to make a difference.

> Change Line 74 to: select(nil, nil, nil, 5)

This looks like a reasonable change. I'm guessing the 4th param relates to timeout.

**Comment by unKn0wnUser453** [comment]
Sometimes depth doesn't work especially if the TFTP root was C:\
I tested it with the following changes and it only works when depth is 0. I tried depth = 1 and it does not work. The overwriting of inputs should either be a flag (set by default) or removed entirely.

**Comment by bcoles** [comment]
I've create a PR #13456 which addresses these concerns among a bunch of other changes.

I wasn't able to reproduce the depth issue. It is rare that exploitation of a directory traversal vulnerability fails due to traversing too high in the directory structure, requiring the exact `depth` to be specified. I was able to exploit this issue with a depth of `10` when the root directory was configured as `C:\\` and `C:\\temp`. Regardless, I've updated the module to support a depth of zero.

While performing tests, I had one failed exploitation attempt, which was most likely due to an insufficient wait time between uploading the EXE and uploading the MOF. As per your suggested patch, I've bumped the `select` call from `1` to `3`. No failures were encountered following this change.
